drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -2314,6 +2314,12 @@ RID RenderingDeviceVulkan::texture_create_shared_from_slice(const TextureView &p
 		image_view_create_info.viewType = VK_IMAGE_VIEW_TYPE_2D_ARRAY;
 	}
 
+	if (p_slice_type == TEXTURE_SLICE_2D) {
+		texture.type = TEXTURE_TYPE_2D;
```

Comment:
@clayjohn @JFonS @reduz I ran into issues with some of the checking in various places that when creating a slice of a `TEXTURE_TYPE_2D_ARRAY`, the texture.type of the slice was still identified as an array, even though it no longer is.

This fixes this and seems to work but wanted to point it out in case you guys in case this has effects I havent thought off.